### PR TITLE
Add a license-checking job

### LINKED ISSUE #801: Add dependency license scanning linting
**What would you like to be added**:

License scanning paralleling the CNCF periodic license scanning, to make sure that all of our code and dependencies use a CNCF-approved license.

Some CNCF scanning details are here, but need to dig for more details about our tooling options:

https://github.com/kubernetes/steering/issues/57

**Why is this needed**:

To ease integration with other CNCF projects.

### PR DESCRIPTION
**Pull request by skitt**
This uses https://github.com/uw-labs/lichen, which is closer to our
requirements for CNCF validation than
https://github.com/mitchellh/golicense and performs validation using
local information.

The references for the Allowlist and the approved exceptions are
described in the lichen configuration file.

Depends on https://github.com/submariner-io/submariner/pull/1344
Fixes: #801
Signed-off-by: Stephen Kitt <skitt@redhat.com>

### CODE REVIEW

**Comment by skitt** [comment]
The failure is expected, because `go-ipify`’s license (the “Unlicense”) isn’t in the approved list and it doesn’t have an exception; this is something I assume we’ll need to fix, either by requesting an exception or by using some other tool (which fits in with the discussions around determining public IP addresses anyway).

File: Makefile [hunk 1]
```diff
@@ -85,12 +85,16 @@ IMAGES_ARGS = --platform $(subst $(space),$(comma),$(foreach arch,$(subst $(comm
 
 build: $(foreach arch,$(subst $(comma),$(space),$(ARCHES)),$(foreach binary,$(BINARIES),bin/linux/$(call gotodockerarch,$(arch))/$(binary)))
 
+licensecheck: BUILD_ARGS=--debug
+licensecheck: build
+	lichen -c .lichen.yaml $(foreach arch,$(subst $(comma),$(space),$(ARCHES)),$(foreach binary,$(BINARIES),bin/linux/$(call gotodockerarch,$(arch))/$(binary)))
```
Comment (skitt) [inline Makefile]: The construction of the list of binaries should be shared with the build target.

File: Makefile [hunk 2]
```diff
@@ -85,12 +85,16 @@ IMAGES_ARGS = --platform $(subst $(space),$(comma),$(foreach arch,$(subst $(comm
...
+
 ci: validate unit build images
 
 $(TARGETS): vendor/modules.txt
 	./scripts/$@
 
-.PHONY: $(TARGETS) build ci images unit validate
+.PHONY: $(TARGETS) build ci images unit validate list-binaries
```
Comment (skitt) [inline Makefile]: This is obviously wrong.

**Comment by dfarrell07** [comment]
The tool and PR overall look great, ACK to your self-review -1 for now.

File: .lichen.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  - "MIT"
+  - "ISC"
+  - "Python-2.0"
+  - "PostgreSQL"
+  - "X11"
+  - "Zlib"
+
+override:
+  - path: "github.com/rdegges/go-ipify"
+    licenses: ["Unlicense"]
```
Comment (mangelajo) [inline .lichen.yaml]:
```suggestion
```

We can remove this now.

Comment (skitt) [reply .lichen.yaml]: So we can, thanks!

**Review by dfarrell07 — APPROVED** [review]
Nice! :partying_face:

**Review by tpantelis — APPROVED** [review]

**Pull request merged**